src/util.js

```diff
@@ -46,7 +46,7 @@ export function getMatchingValue(objectLike, stringOrRegex) {
 }
 
 export function getMatchingValuesByAll(arrayLike, arrayOfStringOrRegex, propertyToMatch) {
-	if (!Array.isArray(arrayOfStringOrRegex) || !propertyToMatch) {
+	if (!Array.isArray(arrayLike) || !Array.isArray(arrayOfStringOrRegex) || !propertyToMatch) {
```

Comment:
If `arrayLike` is not an array then just return an empty array the way we would if any other input is `undefined`/`null`.